src/atom.cpp

```diff
@@ -2062,28 +2062,63 @@ void Atom::add_callback(int flag)
   for (ifix = 0; ifix < modify->nfix; ifix++)
     if (modify->fix[ifix] == NULL) break;
 
-  // add callback to lists, reallocating if necessary
+  // insert callback into sorted lists, reallocating if necessary
+  // sorting is required in cases where fixes were replaced as it ensures atom
+  // data is read/written/transfered in the same order that fixes are called
 
   if (flag == 0) {
     if (nextra_grow == nextra_grow_max) {
       nextra_grow_max += DELTA;
       memory->grow(extra_grow,nextra_grow_max,"atom:extra_grow");
     }
-    extra_grow[nextra_grow] = ifix;
+
+    for(i = 0; i < nextra_grow; i++)
+        if(ifix < extra_grow[i]) break;
+
+    if(i == nextra_grow) {
```

Comment:
I thought you were going to do this with
std::sort() rather than massage the list
and insertion by hand?   The sort() is just a one-liner.

Reply:
I liked std::sort since it's easier to parse but it requires including \<algorithm\>. Are there any objections to adding more header files to main classes? If not, I can easily make the swap. I tested and confirmed both options work.

Reply:
@jtclemm adding `#include <algorithm>` is not a problem for as long as it is in the atom.cpp file only. The concern about avoiding include statements is about the header files only, since some files need to include nearly all of them at the same time (directly or indirectly) and then we have had in the past too many conflicts and collisions.
With the recent changes to modernize some of the code, we are a bit more liberal with C++ standard headers, so common containers like `<string>`, `<vector>`, `<map>` may be used as well.

Reply:
Great, that's good to know. I'll make the change.

Thank you for the input.